### PR TITLE
Add types for extensions.istio.io

### PR DESCRIPTION
**Pull request by dgn**
Adding codegen for types introduced in https://github.com/istio/api/pull/1940

### CODE REVIEW

File: Makefile.core.mk [hunk 1]
```diff
@@ -77,7 +79,7 @@ ifeq ($(IN_BUILD_CONTAINER),1)
 	# code into ./<package>, e.g. ./istio.io/client-go/pkg/apis/...  To work
 	# around this, we'll just let k8s generate the code where it wants and copy
 	# back to where it should have been generated.
-	move_generated=cp -r $(GOPATH)/src/$(kube_base_output_package)/ . && rm -rf $(GOPATH)/src/$(kube_base_output_package)/
+	move_generated=go mod download && cp -r $(GOPATH)/src/$(kube_base_output_package)/ . && rm -rf $(GOPATH)/src/$(kube_base_output_package)/
```
Comment (dgn) [inline Makefile.core.mk:82]: without adding the `go mod download` here I couldn't get this to work in a non-prow environment (aka my laptop)

**Review by shamsher31 — APPROVED** [review]
LGTM.

**Pull request merged**